server/app/controllers/fusor/api/v21/deployments_controller.rb

```diff
@@ -64,6 +65,15 @@ def deploy
       end
     end
 
+    def validate
+      status = @deployment.valid? ? 200 : 422
+
+      render status: status, json: {
+        :errors => @deployment.errors.full_messages,
```

Comment:
@daviddavis, this may give an error if `@deployment.errors` is nil. You may need to ad `if @deployment.errors.present?`. Did you check?